ice/classes/sanger_analysis.py

```diff
@@ -414,26 +419,19 @@ def _generate_edit_proposals(self):
             deletion_befores = list(range(self.indel_max_size + 1))
             deletion_afters = list(range(self.indel_max_size + 1))
 
-            # if there is a donor, remove deletions that are the same size as hdr deletion as they may
-            # interfere with the inference
-            if self.donor_odn and self.donor_alignment.hdr_indel_size in deletion_befores:
-                deletion_befores.remove(self.donor_alignment.hdr_indel_size)
-                deletion_afters.remove(self.donor_alignment.hdr_indel_size)
-
             for deletion_before in deletion_befores:
                 for deletion_after in deletion_afters:
+                    if self._should_skip_proposal(deletion_before + deletion_after):
+                        continue
```

Comment:
_should_skip_proposal treats deletions and insertions the same way?
self.donor_alignment.hdr_indel_size  ==> if there is a 10bp deletion, is this value 10 or -10?

If there is a 10bp insertion, would it also throw out the 10bp deletion edit proposal?

Maybe have some more comments in the docstring for _should_skip_proposal

Reply:
Good catch. Don't see any reason you would want to throw out both the insertion and deletion. hdr_indel_size is negative for deletions (added a deletion only test case to hdr_indel_size).